src/__tests__/componentRenderer.test.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import styled from 'styled-components'
+import { componentRenderer } from '../componentRenderer'
+
+const modules = {
+  Foo: () => <div>foo</div>,
+  Bar: () => <div>bar</div>,
+  Baz: ({ name }) => <div>{name}</div>
+}
+
+describe('componentRenderer', () => {
+  describe('server-side', () => {
+    it('returns mapped components', () => {
+      const { components } = componentRenderer({
+        mode: 'server',
+        modules
+      })
+
+      expect(Object.keys(components)).toEqual(['Foo', 'Bar', 'Baz'])
+    })
+
+    it('renders component html', () => {
+      const renderQueue = []
+
+      const { components } = componentRenderer({
+        mode: 'server',
+        modules
+      })
+
+      expect(components.Foo()).toContain('foo')
+      expect(components.Bar()).toContain('bar')
+      expect(components.Baz({ name: 'baz' })).toContain('baz')
+    })
+
+    it('injects DOM id mount points', () => {
+      const { components } = componentRenderer({
+        mode: 'server',
+        modules
+      })
+
+      expect(components.Foo()).toContain('id=stitch-component-')
+    })
+
+    it('renders component css', () => {
+      const Wrapper = styled.div`
+        background-color: purple;
+      `
+
+      const { components } = componentRenderer({
+        mode: 'server',
+        modules: {
+          Wrapper
+        }
+      })
+
+      expect(components.Wrapper()).toContain('background-color:purple;')
```

Comment:
With SC server rendering does the `background-color` here get attached as an inline style on the `div`? I guess I was surprised that this assertion works.

Reply:
Ah no, it inserts `<style>` tags at the call-site